### PR TITLE
Added automatic deploy to pypi

### PR DESCRIPTION
**Pull request by Lobosque**
**What current issue(s) does this address, or what feature is it adding?**

Automatically releases new versions of `neo-python` to pypi.org (Relates to #64 )

**How did you solve this problem?**

Deploying through travis-ci

**How did you make sure your solution works?**

I created a new project on pypi to make sure this solution works (I will delete it after this is done):
https://travis-ci.org/Lobosque/neo-python
https://pypi.org/project/neo-python-TESTDEPLOY/

**Are there any special changes in the code that we should be aware of?**

SECURE_PASSWORD_HERE is a placeholder. @localhuman or anyone who controls the `CityOfZion` pypi account should run `travis encrypt --add deploy.password` to add the password to `.travis.yml`
Details here: https://docs.travis-ci.com/user/deployment/pypi/

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [ ] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,3 +24,15 @@ script:
...
+
+deploy:
+  provider: pypi
+  distributions: sdist
+  user: CityOfZion
+  password:
+    secure: SECURE_PASSWORD_HERE
+  on:
+    tags: true
+    all_branches: true
```
Comment (metachris) [inline .travis.yml]: it should only do this on tags, not all branches. See this for reference https://github.com/CityOfZion/neo-python-core/blob/master/.travis.yml#L36

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1,10 @@
++include README.md
++include prompt.py
++
```
Comment (metachris) [inline MANIFEST.in]: doesn't it need to include `neo` and all subdirectories too here?

Comment (Lobosque) [reply MANIFEST.in]: Thanks, you are right. Just fixed this.

**Comment by metachris** [comment]
You're not supposed to do version bumps in PRs. Please leave it at `0.4.11-dev`

File: .travis.yml [hunk 3]
```diff
@@ -24,3 +24,15 @@ script:
...
+    repo: CityOfZion/neo-python
+python: 3.5
```
Comment (metachris) [inline .travis.yml]: remove this line

**Comment by Lobosque** [comment]
This was a mistake, I thought I was in another branch to check removing the `all_branches` configuration. I already forced a push to fix it.

**Comment by metachris** [comment]
Ok. Please note the open comments inline.

File: .travis.yml (hunk 1 above)
Comment (Lobosque) [reply .travis.yml]: In my experience adding only `tags: true` did not trigger the deploy.
I found an issue on travis from a few years ago where people reported to have the same issue.
I'm gonna try once again without it and see how it goes.

Comment (metachris) [reply .travis.yml]: the linked travis.yml file above works. please change this PR to use the exact same way as neo-python-core.

File: .travis.yml [hunk 4]
```diff
@@ -24,3 +24,15 @@ script:
 
 after_success:
     coveralls
+
+deploy:
+  true:
+    tags: true
+    python: 3.5
+    repo: CityOfZion/neo-python-core
```
Comment (metachris) [inline .travis.yml]: now you reference the wrong repository

File: MANIFEST.in [hunk 5]
```diff
@@ -0,0 +1,11 @@
+include README.md
+include prompt.py
```
Comment (metachris) [inline MANIFEST.in]: what we also need to include is `bootstrap.py`, the `protocol.*.json` files, and `reencrypt_wallet.py`

**Comment by metachris** [comment]
Added more comments.

In general I think auto-deployments are great, btw. Thanks for pushing on this. 

Did you make sure that in the Manifest, the include neo part also includes all sub- and sub-subdirectories? Please double-check,

File: MANIFEST.in (hunk 5 above)
Comment (Lobosque) [reply MANIFEST.in]: fixed

**Comment by Lobosque** [comment]
@metachris just double-checked and all the sub-folders (and sub-sub-folders) are being included!

File: .travis.yml [hunk 6]
```diff
@@ -24,3 +24,15 @@ script:
 
 after_success:
     coveralls
+
+deploy:
+  true:
+    tags: true
+    python: 3.5
```
Comment (localhuman) [inline .travis.yml]: going forward we are on python 3.6

Comment (Lobosque) [reply .travis.yml]: @localhuman fixed!

**Pull request merged**